src/safeStringify.ts

```diff
@@ -5,13 +5,11 @@ export const safeStringify = (
   replacer?: (key: string, value: any) => any,
   space?: string | number
 ): string => {
-  if (!value || typeof value === 'string') return value;
```

Comment:
I'm fine with changing it to 
```ts
if (value === undefined || typeof value === 'string') return value;
```
but I truly believe we should be aligned with `JSON.stringify`

Reply:
If we just needed circular dependency handling then we can use `fastStringify` directly. This is meant to protect against double stringifying - otherwise we need to put this protect against double stringifying in many places across our code bases.

Reply:
Maybe a comment on this line would help to make this clear to future folks?

Reply:
I'd say `"foo"` is *not* a stringified value, at least it's not a valid JSON.
To illustrate:
`JSON.parse('true'); // true`
`JSON.parse('"true"'); // 'true'`
The differencies are subtle, but do matter.
If I pass any string and stringify it, I'd expect the return value to be a valid JSON that is parseable by `JSON.parse`.
The current implementation does not meet the requirement.

Reply:
If we want to avoid double stringifying we make the following check:
```ts
const isStringified = (val: unknown) => {
  try {
    return typeo JSON.parse(val) === 'string';
  } catch {
    return false;
  }
}
```

Reply:
> If I pass any string and stringify it, I'd expect the return value to be a valid JSON that is parseable by JSON.parse

🤔 hmm ok the `isStringified` proposal is nice. Can you make sure to only call `isStringified` when relevant (when value passed into `safeStringify` is a string already? Because it DOES add an extra parse which can hurt performance, but should only matter when the incoming value is a string, which does not happen often.

Reply:
>  when relevant (when value passed into safeStringify is a string already? 

Yup!